### PR TITLE
Trigger: Improve error handling and reporting

### PR DESCRIPTION
**Pull request by tunnell**
Improve error handling by implementing custom exceptions in Trigger logic and better reporting/addressing errors.

### CODE REVIEW

File: pax/exceptions.py [hunk 1]
```diff
@@ -1,19 +1,25 @@
 
-class PulseBeyondEventError(Exception):
+class PaxException(Exception):
+    pass
+
+class PulseBeyondEventError(PaxException):
```
Comment (tunnell) [inline pax/exceptions.py]: __[flake8]__
* [E302] expected 2 blank lines, found 1

File: pax/exceptions.py [hunk 2]
```diff
@@ -1,19 +1,25 @@
...
-class MaybeOldFormatException(Exception):
+class MaybeOldFormatException(PaxException):
     pass
 
 
-class InvalidConfigurationError(Exception):
+class InvalidConfigurationError(PaxException):
     pass
+
+class TriggerGroupSignals(PaxException):
```
Comment (tunnell) [inline pax/exceptions.py]: __[flake8]__
* [E302] expected 2 blank lines, found 1

File: pax/exceptions.py [hunk 3]
```diff
@@ -1,19 +1,25 @@
...
+    pass
```
Comment (tunnell) [inline pax/exceptions.py]: __[flake8]__
* [W292] no newline at end of file

File: bin/event-builder [hunk 4]
```diff
@@ -174,59 +175,26 @@ def main():
...
-                               "sender": "trigger",
-                               }
-                alerts.insert(alert_doc)
-
-                # Mark run as errored
-                runs.update({'name': run_doc['name'], 'detector': args.detector},
-                            {'$set': {status_name: 'error'}})
+        except (pymongo.errors.ServerSelectionTimeoutError,
+                PaxException) as e:
+            # There was a mongo timeout. This probably means the connection to
```
Comment (JelleAalbers) [inline bin/event-builder]: .. or a different, known exception was caught

File: pax/trigger_plugins/SaveSignals.py [hunk 5]
```diff
@@ -52,7 +53,7 @@ def group_signals(signals, event_ranges, signal_indices_buffer, is_in_event):
         if not in_event:
             if signal['left_time'] >= event_ranges[current_event, 0]:
                 if signal['left_time'] > event_ranges[current_event, 1]:
-                    raise ValueError("Error during signal grouping: event without signals??")
+                    raise TriggerGroupSignals("Error during signal grouping: event without signals??")
```
Comment (JelleAalbers) [inline pax/trigger_plugins/SaveSignals.py]: I may propose some name changes (e.g. TriggerLogicError / TriggerSignalGroupingError) and add a few more custom exceptions at other points. Ok now of course.

**Comment by JelleAalbers** [comment]
Looks good, even though the specific error causing the crash should be fixed now, this makes the handling clearer and lets us add more other exceptions. Just merge as soon as it builds (I can do it too), we can look at comments / docstrings later.

**Comment by JelleAalbers** [comment]
I fixed the flake8 errors and expanded some comments. While doing so my fingers betrayed me and made two minor changes, hope you don't mind (otherwise just hit revert, or ask me to revert, and we can discuss first):
 - Write the exception to the log before asking for a new runs db connection (to send the alert to the website). This ensures the exception is logged even if there are problems with the runs db.
 - Refactor a global constant to the top of the file (and uppercase it).

**Pull request merged**